pkg/generate/app/cmd/newapp.go

```diff
@@ -386,6 +425,19 @@ func (c *AppConfig) Run(out io.Writer) (*AppResult, error) {
 
 	objects = app.AddServices(objects)
 
+	templates, err := c.buildTemplates(components, app.Environment(parameters))
+	if err != nil {
+		return nil, err
+	}
+	for _, t := range templates {
+		templateResolver := c.templateResolver.(app.TemplateResolver)
+		items, err := t.Process(templateResolver.TemplateConfigsNamespacer, templateResolver.Namespaces[0])
```

Comment:
This is breaking encapsulation of process and the resolver, and doing buildTemplates up above but calling process here doesn't make this easier to read.  buildTemplates should include this entire block